### PR TITLE
Use Metal Detector when boss is dying in 10 seconds or less

### PR DESCRIPTION
**Pull request by Xinayder**
I haven't tested it yet. It should work, although I'd recommend using Math.floor on the return value of `predictBossDeath();`.

In case this is not tested and merged until I'm online again, I'll try to fix this if it isn't working.

### CODE REVIEW

**Comment by ozerugae** [comment]
Should probably use Treasure too

Didn't seem to work, it (or something) called it in too early - right when the boss came up. Gonna check again in 40-ish minutes if it worked

**Comment by ozerugae** [comment]
@RockyTV 

Found the issue:

###############
Enemy: [object Object]
Enemy Boss Health: 14895205357500000
Team DPS: 0
Secs to Boss Death: 0
###############

When the boss level initially starts, the dps is zero ;)

File: autoPlay.user.js [hunk 1]
```diff
@@ -691,11 +702,11 @@ function useMetalDetectorIfRelevant() {
 		var enemy = g_Minigame.m_CurrentScene.GetEnemy(g_Minigame.m_CurrentScene.m_rgPlayerData.current_lane, g_Minigame.m_CurrentScene.m_rgPlayerData.target);
 		// check if current target is a boss, otherwise we won't use metal detector
 		if (enemy && enemy.m_data.type == ENEMY_TYPE.BOSS) {
-			var enemyBossHealthPercent = enemy.m_flDisplayedHP / enemy.m_data.max_hp;
+			var secsToBossDeath = predictBossDeath();
 
-			if (enemyBossHealthPercent >= 0.9) { // We want sufficient time for the metal detector to be applicable
+			if (secsToBossDeath <= 10) { // We want sufficient time for the metal detector to be applicable
```
Comment (ozerugae) [inline autoPlay.user.js:707]: Maybe 

``` javascript
    if (secsToBossDeath <= 10 && secsToBossDeath > 1)
```

will be sufficient

**Comment by Xinayder** [comment]
Closed, made a new PR to track @chauffer's work.

**Pull request closed without merging**